### PR TITLE
refine the nvidia-smi query duration [fix]

### PR DESCRIPTION
**Pull request by YanjieGao**
change duration to 30 seconds and configurable

### CODE REVIEW

File: prometheus/exporter/gpu_exporter.go [hunk 1]
```diff
@@ -166,12 +167,10 @@ func main() {
         log.Print("no gpu")   
         return   
     }
-    var index = 1
  
-    for a := 0; a < 1; {
+    for ; ; {
```
Comment (fanyangCS) [inline prometheus/exporter/gpu_exporter.go]: > ; ; [](start = 7, length = 4)

remove

File: prometheus/exporter/gpu_exporter.go [hunk 2]
```diff
@@ -154,6 +154,7 @@ func writeMetricsToFile(index int, logDir string) {
 func main() {
     logDir := flag.String("gpu_log_dir", "/datastorage/prom-exporters", "log written dir")   
     mode := flag.String("flag", "gpu", "gpu server")
+    sleepTime := flag.Int("sleepTime", 30, "cmd sleep duration")
```
Comment (fanyangCS) [inline prometheus/exporter/gpu_exporter.go]: >sleepTime [](start = 4, length = 9)

should also change dockerfile entry point?

File: prometheus/exporter/gpu_exporter.go [hunk 3]
```diff
@@ -154,6 +154,7 @@ func writeMetricsToFile(index int, logDir string) {
 func main() {
     logDir := flag.String("gpu_log_dir", "/datastorage/prom-exporters", "log written dir")   
     mode := flag.String("flag", "gpu", "gpu server")
```
Comment (fanyangCS) [inline prometheus/exporter/gpu_exporter.go]: >mode [](start = 4, length = 4)

do we need a default mode? and default sleeptime?

Comment (YanjieGao) [reply prometheus/exporter/gpu_exporter.go]: Yes and the entry point will override this.

File: prometheus/exporter/gpu_exporter.go (hunk 2 above)
Comment (YanjieGao) [reply prometheus/exporter/gpu_exporter.go]: added

File: prometheus/exporter/gpu_exporter.go (hunk 1 above)
Comment (fanyangCS) [reply prometheus/exporter/gpu_exporter.go]: [https://www.cyberciti.biz/faq/golang-for-loop-examples/](https://www.cyberciti.biz/faq/golang-for-loop-examples/), search for "Infinite for loop syntax"

---
In reply to: [160069097](https://github.com/Microsoft/pai/pull/188#discussion_r160069097) [](ancestors = 160069097)

File: prometheus/exporter/gpu_exporter.go (hunk 3 above)
Comment (YanjieGao) [reply prometheus/exporter/gpu_exporter.go]: remove ;; and test docker build and run

**Pull request merged**